differ/renderer.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright © 2021 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package differ
+
+import (
+	"bytes"
+	"encoding/json"
+	httputils "github.com/RedHatInsights/insights-operator-utils/http"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/RedHatInsights/ccx-notification-service/conf"
+	"github.com/RedHatInsights/ccx-notification-service/types"
+	utypes "github.com/RedHatInsights/insights-results-types"
+	"github.com/rs/zerolog/log"
+)
+
+func renderReportsForCluster(
+	config conf.DependenciesConfiguration,
+	clusterName types.ClusterName,
+	reports []types.ReportItem,
+	ruleContent types.RulesMap) ([]types.RenderedReport, error) {
+
+	rendererURL := httputils.SetHTTPPrefix(config.TemplateRendererServer + config.TemplateRendererEndpoint)
+
+	req, err := createTemplateRendererRequest(ruleContent, reports, clusterName, rendererURL)
+	if err != nil {
+		log.Error().Err(err).Msg("Request to content template renderer could not be created")
+		return nil, err
+	}
+
+	body, err := httputils.SendRequest(req, 10*time.Second)
+	if err != nil {
+		log.Error().Err(err).Msg("Request to content template renderer could not be processed")
+		return nil, err
+	}
+
+	var receivedResult types.TemplateRendererOutput
+
+	err = json.Unmarshal(body, &receivedResult)
+	if err != nil {
+		log.Error().Err(err).Msg("Error trying to decode template renderer output from received answer")
+		os.Exit(ExitStatusFetchContentError)
```

Comment:
why `os.Exit` instead of returning an error?

Reply:
You are right, replaced with `return nil, err`